integration-tests/container/processes-listening-on-ports/listening-endpoint-child-process-no-fork.c

```diff
@@ -0,0 +1,70 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+int main(int argc, char* argv[]) {
+  int sockfd;
+  bool is_parent = true;
+
+  // Check if sockfd is provided as a command line argument
+  if (argc == 2) {
+    sockfd = atoi(argv[1]);
+    is_parent = false;
+  } else {
+    // Create a socket if sockfd is not provided as an argument
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
```

Comment:
This part about listening on a port seems to be repeated between the tests, what about reusing it?

Reply:
Good idea. Will do.